src/lib.rs

```diff
@@ -1539,7 +1521,37 @@ impl Build {
             }
         }
 
-        let output = cmd.current_dir(out_dir).output()?;
+        // Use stdin compilation for all flag tests to avoid issues with
+        // language-specific flags like -xassembler-with-cpp.
+        // See: https://github.com/rust-lang/cc-rs/issues/359
+        let flag_str = flag.to_str().unwrap_or("");
```

Comment:
I think using to_string_lossy here is better:

```suggestion
        let flag_str = flag.to_string_lossy();
```